server/datastore/mysql/packs.go

```diff
@@ -316,20 +316,33 @@ func replacePackTargetsDB(ctx context.Context, tx sqlx.ExecerContext, pack *flee
 
 func loadPackTargetsDB(ctx context.Context, q sqlx.QueryerContext, pack *fleet.Pack) error {
 	var targets []fleet.PackTarget
-	sql := `SELECT * FROM pack_targets WHERE pack_id = ?`
+	sql := `
+	SELECT pt.*,
+		COALESCE(h.hostname, '') AS host_name,
+		COALESCE(t.name, '') AS team_name,
+		COALESCE(l.name, '') AS label_name
+	FROM pack_targets pt
+		LEFT JOIN hosts h ON (pt.target_id = h.id)
+		LEFT JOIN teams t ON (pt.target_id = t.id)
+		LEFT JOIN labels l ON (pt.target_id = l.id)
```

Comment:
```suggestion
		LEFT JOIN hosts h ON (pt.target_id = h.id AND pt.type=1)
		LEFT JOIN teams t ON (pt.target_id = t.id AND pt.type=2)
		LEFT JOIN labels l ON (pt.target_id = l.id AND pt.type=0)
```

Otherwise, you might match and have a target have a host name and a team name.

Reply:
Nit: use `fleet.TargetLabel`, `fleet.TargetHost` and `fleet.TargetTeam` values (with `pt.type = ?`).